src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.h

```diff
@@ -0,0 +1,662 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/StatusResponse.h>
+#include <app/reporting/reporting.h>
+
+#include <app/SafeAttributePersistenceProvider.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/support/TypeTraits.h>
+#include <protocols/interaction_model/StatusCode.h>
+#include <unordered_set>
+#include <vector>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CameraAvStreamManagement {
+
+using VideoStreamStruct            = Structs::VideoStreamStruct::Type;
+using AudioStreamStruct            = Structs::AudioStreamStruct::Type;
+using SnapshotStreamStruct         = Structs::SnapshotStreamStruct::Type;
+using AudioCapabilitiesStruct      = Structs::AudioCapabilitiesStruct::Type;
+using VideoSensorParamsStruct      = Structs::VideoSensorParamsStruct::Type;
+using SnapshotParamsStruct         = Structs::SnapshotParamsStruct::Type;
+using VideoResolutionStruct        = Structs::VideoResolutionStruct::Type;
+using ViewportStruct               = Structs::ViewportStruct::Type;
+using RateDistortionTradeOffStruct = Structs::RateDistortionTradeOffPointsStruct::Type;
+using SnapshotParamsStruct         = Structs::SnapshotParamsStruct::Type;
+
+constexpr uint8_t kMaxSpeakerLevel          = 254;
+constexpr uint8_t kMaxMicrophoneLevel       = 254;
+constexpr uint16_t kMaxImageRotationDegrees = 359;
+constexpr uint8_t kMaxChannelCount          = 8;
+constexpr uint8_t kMaxImageQualityMetric    = 100;
+constexpr uint16_t kMaxFragmentLenMaxValue  = 65500;
+// Conservative room for other fields(resolution + codec) in
+// capture snapshot response. TODO: Make a tighter bound.
+constexpr size_t kMaxSnapshotImageSize = kMaxLargeSecureSduLengthBytes - 100;
+
+constexpr size_t kViewportStructMaxSerializedSize =
+    TLV::EstimateStructOverhead(sizeof(uint16_t), sizeof(uint16_t), sizeof(uint16_t), sizeof(uint16_t));
+
+// The number of possible values of StreamUsageEnum.
+constexpr size_t kNumOfStreamUsageTypes = 4;
+
+// StreamUsageEnum + Anonymous tag ( 1 byte ).
+// Assumes min-size encoding ( 1 byte ) for the integer.
```

Comment:
```suggestion
// StreamUsageEnum + Anonymous tag (1 byte).
// Assumes min-size encoding (1 byte) for the integer.
```

"parentheticals" and "parentheses" are not the same thing.